boot-qemu.sh

```diff
@@ -237,14 +238,19 @@ function setup_qemu_args() {
                 -machine "virt,gic-version=max"
             )
             if [[ "$(uname -m)" = "aarch64" && -e /dev/kvm ]] && ${KVM}; then
-                QEMU_ARCH_ARGS+=(-enable-kvm)
+                QEMU_ARCH_ARGS+=(
+                    -enable-kvm
+                    -smp "${SMP:-$(get_default_smp_value)}"
+                )
             else
                 QEMU_ARCH_ARGS+=(-machine "virtualization=true")
             fi
             if ${DEBIAN}; then
                 # Booting is so slow without these
                 QEMU_RAM=2G
-                QEMU_ARCH_ARGS+=(-smp "${SMP:-4}")
+                if ! echo "${QEMU_ARCH_ARGS[*]}" | grep -q smp; then
+                    QEMU_ARCH_ARGS+=(-smp "${SMP:-4}")
```

Comment:
how come you use `4` here still rather than `get_default_smp_value`?

Reply:
This block will now only run under TCG (i.e., no virtualization). In the past, I have experienced some sluggishness with a high number of vCPUs when running with TCG, so I opted to leave this block alone. I could add a comment if that would help for the future?

Reply:
Maybe the comment above could be re-worded. Doesn't matter to me.

Reply:
I have added a comment. Scream if you disagree with how it is worded!